File: 513-find-bottom-left-tree-value/find-bottom-left-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    public void solve(TreeNode root, int lvl, int[] maxLvl, int[] ans)
    {
        if(root == null)
            return;
        
        if(lvl > maxLvl[0])
        {
            ans[0] = root.val;
            maxLvl[0] = lvl;
        }

        solve(root.left,lvl+1,maxLvl,ans);
        solve(root.right,lvl+1,maxLvl,ans);
    }

    public int findBottomLeftValue(TreeNode root) {
        
        if(root == null)
            return -1;
        
        int lvl = 0;
        int [] max_lvl = {0};

        int [] ans = {root.val};

        solve(root,lvl,max_lvl,ans);

        return ans[0];
    }
}

File: 513-find-bottom-left-tree-value/find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void solve(int &maxLvl,int lvl,int &ans, TreeNode* root)
    {
        if(!root)
            return;
        
         if(lvl > maxLvl)
        {
            ans = root->val;
            // cout<<ans<<" : ";
            maxLvl = lvl;
        }

        solve(maxLvl,lvl+1,ans,root->left);
        
        solve(maxLvl,lvl+1,ans,root->right);
    }
    int findBottomLeftValue(TreeNode* root) {
        

        if(!root)
            return -1;
        
        int ans = root->val;
        int lvl = 0 , maxLvl = 0;

        solve(maxLvl,lvl,ans,root);

        return ans;

    }
};